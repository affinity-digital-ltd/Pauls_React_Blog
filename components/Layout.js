import { Component } from 'react'

import Header from '../components/Header'
import Footer from '../components/Footer'
import '../scss/application.scss'

class Layout extends Component {
  render () {
    return <div className='container'>
      <Header />
      { this.props.children }
      <Footer />
    </div>
  }
}

export default Layout
