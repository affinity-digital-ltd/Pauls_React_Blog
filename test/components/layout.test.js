import Layout from '../../components/Footer'

describe('<Layout />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <Layout />
    )

    expect(component).toMatchSnapshot()
  })
})
