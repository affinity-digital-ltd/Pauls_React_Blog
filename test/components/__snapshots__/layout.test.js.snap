// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Layout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SideEffect(Head)>
          <title>
            Paul Brighton's Web Development Blog
          </title>
          <meta
            content="A blog by Paul Brighton all about web development. It includes articles about HTML, CSS, JavaScript and Ruby on Rails plus other tips on web design and developing."
            name="description"
          />
          <meta
            content="width=device-width, initial-scale=1"
            name="viewport"
          />
          <link
            href="/static/images/favicon.ico"
            rel="shortcut icon"
            type="image/x-icon"
          />
        </SideEffect(Head)>,
        <Header />,
        undefined,
        <Footer />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <title>
              Paul Brighton's Web Development Blog
            </title>,
            <meta
              content="A blog by Paul Brighton all about web development. It includes articles about HTML, CSS, JavaScript and Ruby on Rails plus other tips on web design and developing."
              name="description"
            />,
            <meta
              content="width=device-width, initial-scale=1"
              name="viewport"
            />,
            <link
              href="/static/images/favicon.ico"
              rel="shortcut icon"
              type="image/x-icon"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Paul Brighton's Web Development Blog",
            },
            "ref": null,
            "rendered": "Paul Brighton's Web Development Blog",
            "type": "title",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "A blog by Paul Brighton all about web development. It includes articles about HTML, CSS, JavaScript and Ruby on Rails plus other tips on web design and developing.",
              "name": "description",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "width=device-width, initial-scale=1",
              "name": "viewport",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "href": "/static/images/favicon.ico",
              "rel": "shortcut icon",
              "type": "image/x-icon",
            },
            "ref": null,
            "rendered": null,
            "type": "link",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SideEffect(Head)>
            <title>
              Paul Brighton's Web Development Blog
            </title>
            <meta
              content="A blog by Paul Brighton all about web development. It includes articles about HTML, CSS, JavaScript and Ruby on Rails plus other tips on web design and developing."
              name="description"
            />
            <meta
              content="width=device-width, initial-scale=1"
              name="viewport"
            />
            <link
              href="/static/images/favicon.ico"
              rel="shortcut icon"
              type="image/x-icon"
            />
          </SideEffect(Head)>,
          <Header />,
          undefined,
          <Footer />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <title>
                Paul Brighton's Web Development Blog
              </title>,
              <meta
                content="A blog by Paul Brighton all about web development. It includes articles about HTML, CSS, JavaScript and Ruby on Rails plus other tips on web design and developing."
                name="description"
              />,
              <meta
                content="width=device-width, initial-scale=1"
                name="viewport"
              />,
              <link
                href="/static/images/favicon.ico"
                rel="shortcut icon"
                type="image/x-icon"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Paul Brighton's Web Development Blog",
              },
              "ref": null,
              "rendered": "Paul Brighton's Web Development Blog",
              "type": "title",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "A blog by Paul Brighton all about web development. It includes articles about HTML, CSS, JavaScript and Ruby on Rails plus other tips on web design and developing.",
                "name": "description",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "width=device-width, initial-scale=1",
                "name": "viewport",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "href": "/static/images/favicon.ico",
                "rel": "shortcut icon",
                "type": "image/x-icon",
              },
              "ref": null,
              "rendered": null,
              "type": "link",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
